types/jsreport-core/index.d.ts

```diff
@@ -7,90 +7,107 @@
-		afterRenderListeners: JsReport.Listener;
-		afterTemplatingEnginesExecutedListeners: JsReport.Listener;
-		beforeRenderListeners: JsReport.Listener;
-		documentStore: JsReport.DocumentStore;
-		initializeListeners: JsReport.Listener;
-		logger: JsReport.Logger;
-		validateRenderListeners: JsReport.Listener;
-		init(): Promise<void>;
-		render(options: JsReport.RenderOptions): Promise<JsReport.Report>;
-		use(extension: any): any;
-	}
+  type Helpers = string | { [fun: string]: (...args: any[]) => any };
+
+  const enum Engine {
+    None = "none"
+  }
+
+  const enum Recipe {
+    Html = "html"
+  }
+
+  interface Template {
+    content: string;
+    engine: Engine | string;
+    helpers: Helpers;
+    recipe: Recipe | string;
+  }
+
+  interface Request {
+    template: Partial<Template>;
+    options: object
+    data: any;
+  }
+
+  interface Response {
+    content: Buffer;
+    stream: NodeJS.ReadableStream;
+    headers: {
+      [header: string]: string | number | boolean;
+    };
+  }
+
+  interface ListenerCollection {
+    add(
+      type: string,
+      callback: (req: Request, res: Response, err?: any) => Promise<any> | void
+    ): void;
+  } 
+
+  interface Collection {
+    find(query: { [field: string]: any }): Promise<object[]>;
+    update(query: { [field: string]: any }, update: object, options?: object): Promise<any>;
+    remove(query: { [field: string]: any }): Promise<any>;
+    insert(obj: object): Promise<object>;
+  }
+
+  interface DocumentStore {
+    collection(name: string): Collection;
+  }
+
+  interface Reporter {
+    afterRenderListeners: ListenerCollection;
+    afterTemplatingEnginesExecutedListeners: ListenerCollection;
+    beforeRenderListeners: ListenerCollection;
+    documentStore: DocumentStore;
+    initializeListeners: ListenerCollection;
+    // it would be nice to add winston.LoggerInstance here
+    // however adding import winston = require('winston') breaks exported enums
+    logger: any;
+    validateRenderListeners: ListenerCollection;
+    version: string;
+    init(): Promise<Reporter>;
+    render(options: Partial<Request>): Promise<Response>;
+    use(extension: any): Reporter;
+    discover(): Reporter;
+    createListenerCollection(): ListenerCollection;
+  }
+
+  const enum TasksStrategy {
+    DedicatedProcess = "dedicated-process",
+    HttpServer = "http-server",
+    InProcess = "in-process"
+  }
+
+  interface Configuration {
+    autoTempCleanup: boolean;
+    dataDirectory: string;
+    extensionsLocationCache: boolean;
+    loadConfig: boolean;
+    logger: {
+      silent: boolean;
+    };
+    rootDirectory: string;
+    scripts: {
+      allowedModules: string[];
+    };
+    tasks: Partial<{
+      allowedModules: string[] | string;
+      strategy: TasksStrategy;
+    }>;
+    tempDirectory: string;
+  }
+
+  // without exporting enum, it doesn't include the require('jsreport-core') in the test.js for some reason
+  // help welcome
+  export enum Foo {}
```

Comment:
I don't get it, I'm sorry for my bad English.

Reply:
The transpiled output of `tests.ts` looks normally like this

```js
var _this = this;
exports.__esModule = true;
var JsReport = require("jsreport-core");
var JsReportPhantomPdf = require("jsreport-phantom-pdf");
var jsreport = JsReport({
    tasks: {
        strategy: "http-server" /* HttpServer */
    }
});
```

However if you comment out the
```js
// export enum Foo {}
```

The output is like this

```js
var _this = this;
exports.__esModule = true;
var JsReportPhantomPdf = require("jsreport-phantom-pdf");
var jsreport = JsReport({
    tasks: {
        strategy: "http-server" /* HttpServer */
    }
});
```

You see it is missing `var JsReport = require("jsreport-core");`. It doesn't work because of that. I don't understand why.